testing/src/test/scala/org/http4s/Http4sSuite.scala

```diff
@@ -25,6 +25,9 @@ import munit._
 /** Common stack for http4s' munit based tests
   */
 trait Http4sSuite extends CatsEffectSuite with DisciplineSuite with munit.ScalaCheckEffectSuite {
+  // The default munit EC causes an IllegalArgumentException in
+  // BatchExecutor on Scala 2.12.
+  override val munitExecutionContext = Http4sSpec.TestExecutionContext
```

Comment:
This is interesting. I think I had to use `TestExecutionContext` in a few places too

Reply:
I discussed with @RaasAhsan a bit last night, and I think we can fork tests and get rid of some of our specialized test thread pools.  But I'm just trying to land this plane.